share/completions/rclone.fish

```diff
@@ -1,5 +1,7 @@
-set -l rclone_version (rclone version | string match -rg 'rclone v?(.*)' | string split .)
-or return
+set -l rclone version_str (rclone version 2> /dev/null)
+or return 1
+set -l rclone_version (string match -rg 'rclone v?(.*)' "$rclone_version_str" | string split .)
+or set -l rclone_version 99 99  # Could be a dev build, assume a recent version
```

Comment:
can we do this instead

````c++
begin
    rclone completion fish - || rclone completion fish
end 2>/dev/null | source
```

Reply:
If `rclone completion fish -` exits with an error for some unrelated reason on a recent version of `rclone`, `rclone completion fish` would (insanely) try to install the completions to `/etc/fish/completions`.

Reply:
how about

```fish
rclone completion fish (
    if set -l rclone_version (rclone version | string match -rg 'rclone v?(.*)' | string split .) &&
        test "$rclone_version[1]" -lt 1 ||
        test "$rclone_version[1]" -eq 1 &&
        test "$rclone_version[2]" -le 62
        # version is definitely <= 1.62, use the old style
    else
        # assume new
        echo -
    end
) 2>/dev/null | source
```

Reply:
no need for `type -q rclone` since fish already checks that before sourcing the completion

> Actually, I'd find the following more readable

sure yeah that's better